converter/ena/ena_experiment.py

```diff
@@ -0,0 +1,118 @@
+from converter.ena.classes import AttributeType
+from converter.ena.classes.sra_common import PlatformType, RefObjectType, TypeIlluminaModel
+from converter.ena.classes.sra_experiment import Experiment, ExperimentSet, LibraryDescriptorType, LibraryType, SampleDescriptorType, TypeLibrarySelection, TypeLibrarySource, TypeLibraryStrategy
+
+
+class EnaExperiment:
+
+    def __init__(self, hca_data):
+        self.submission = hca_data.submission
+        self.assays = hca_data.submission["assays"]
+
+    def experiment_set(self):
+        experiment_set = ExperimentSet()
+        for assay in self.assays:
+            experiment_set.experiment.append(self.experiment(assay))
+
+        return experiment_set
+
+    def experiment(self, assay):
+
+        sequencing_protocol = assay["sequencing_protocol"]
+        library_preparation_protocol = assay["library_preparation_protocol"]
+        input_biomaterials = assay["input_biomaterials"]
+
+        experiment = Experiment()
+        experiment.experiment_attributes = Experiment.ExperimentAttributes()
+        experiment.experiment_attributes.experiment_attribute.append(AttributeType(tag="Description", value=sequencing_protocol["content"]["protocol_core"]["protocol_description"]))
+
+        experiment.alias = sequencing_protocol["content"]["protocol_core"]["protocol_id"]
+        experiment.title = sequencing_protocol["content"]["protocol_core"]["protocol_name"]
+
+        experiment.design = LibraryType()
```

Comment:
This is an example of toom many implementation details in the function.
I would extract this entire thing to a builder function.

Reply:
Hi @amnonkhen , I have introduced some form of abstraction with EnaExperiment and EnaRun extending the EnaModel class and need to implement the 2 abstract methods: create() and create_set(). Because the actual Experiment and Run classes are generated stud from the XSDs, I can't think of a way at this time to extract there creation to a builder function (the way I am understanding this pattern works). It is more of a conversion that's being done in this class (naming it converter would probably make more sense): HcaData -> EnaExperiment and EnaRun. I'll push the changes I have done, please suggest more idea if you have, these are useful suggestions.